bow.go

```diff
@@ -343,31 +321,43 @@ func (b *bow) NumCols() int {
 	return int(b.Record.NumCols())
 }
 
-func (b *bow) AddCols(series ...Series) (Bow, error) {
-	seriesNbr := len(series)
-	if seriesNbr == 0 {
+func (b *bow) AddCols(seriesSlice ...Series) (Bow, error) {
+	if len(seriesSlice) == 0 {
 		return b, nil
 	}
 
-	bowNumCols := b.NumCols()
-	addedColNames := make(map[string]*interface{}, bowNumCols+seriesNbr)
-	newSeries := make([]Series, bowNumCols+seriesNbr)
+	addedColNames := make(map[string]*interface{}, b.NumCols()+len(seriesSlice))
+	newSeriesSlice := make([]Series, b.NumCols()+len(seriesSlice))
 
 	for colIndex, col := range b.Schema().Fields() {
-		newSeries[colIndex] = Series{
-			Name:  col.Name,
-			Array: b.Record.Column(colIndex),
-		}
+		newSeriesSlice[colIndex] = b.NewSeriesFromCol(colIndex)
 		addedColNames[col.Name] = nil
 	}
 
-	for i, s := range series {
+	for i, s := range seriesSlice {
 		_, ok := addedColNames[s.Name]
 		if ok {
-			return nil, fmt.Errorf("column %q already exists", s.Name)
+			return nil, fmt.Errorf("bow.AddCols: column %q already exists", s.Name)
 		}
-		newSeries[bowNumCols+i] = s
+		newSeriesSlice[b.NumCols()+i] = s
 		addedColNames[s.Name] = nil
 	}
-	return NewBowWithMetadata(b.GetMetadata(), newSeries...)
+
+	return NewBowWithMetadata(b.Metadata(), newSeriesSlice...)
+}
+
+func getValid(arr array.Interface, length int) []bool {
```

Comment:
Function name not clear, what do we get that is valid ?
-> getValid<Something>
Plus you're not getting, but filtering or extracting
I would suggest extractValid<something> or filterValid<something>

Reply:
You're right :) Is 'getValiditySlice' better?

Reply:
Looks good, i read it too fast you're not extracting but getting all indexes validity so that's fine